base_layer/core/src/transactions/key_manager/memory_db_key_manager.rs

```diff
@@ -42,25 +46,26 @@ fn random_string(len: usize) -> String {
         .collect()
 }
 
-pub fn create_memory_db_key_manager_with_range_proof_size(size: usize) -> MemoryDbKeyManager {
-    let connection = DbConnection::connect_url(&DbConnectionUrl::MemoryShared(random_string(8))).unwrap();
+pub fn create_memory_db_key_manager_with_range_proof_size(
+    size: usize,
+) -> Result<MemoryDbKeyManager, KeyManagerServiceError> {
+    let connection = DbConnection::connect_url(&DbConnectionUrl::MemoryShared(random_string(8)))?;
     let cipher = CipherSeed::new();
 
     let mut key = [0u8; size_of::<Key>()];
     OsRng.fill_bytes(&mut key);
     let key_ga = Key::from_slice(&key);
     let db_cipher = XChaCha20Poly1305::new(key_ga);
     let factory = CryptoFactories::new(size);
-
+    key.zeroize();
```

Comment:
Even though this code can't return early before zeroizing the key, I'd recommend using a `Zeroizing` wrapper for `key` to ensure zeroizing always occurs. This would be a hedge against any future updates that could change this behavior, so it's mostly a nit that I think is generally good practice.

Note that you'd need to replace the references with `key.as_mut()` and `key.as_ref()` as appropriate to access the wrapped array.